naga/src/front/wgsl/parse/directive/language_extension.rs

```diff
@@ -60,9 +60,25 @@ impl LanguageExtension {
     }
 }
 
-/// A variant of [`LanguageExtension::Implemented`].
+// A variant of [`LanguageExtension::Implemented`].
+/// Enum of all WGSL extensions that are implemented
+/// WGSL spec.: <https://www.w3.org/TR/WGSL/#language-extensions-sec>
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub(crate) enum ImplementedLanguageExtension {}
+pub enum ImplementedLanguageExtension {}
+
+impl ImplementedLanguageExtension {
+    const ALL: [ImplementedLanguageExtension; 0] = [];
+
+    /// Returns slice of all variants of [`ImplementedLanguageExtension`]
+    pub const fn all() -> &'static [Self] {
+        &Self::ALL
+    }
```

Comment:
nitpick: It would be more future-proof to implement this in terms of `strum::{EnumIter,IntoEnumIterator}`, and return an `impl Iterator<Item = ImplementedLanguageExtension` instead.more maintenance-proof.

Reply:
Oh didn't know we already have strum as dep, will fix this.

Reply:
Actually it's only dev-dep and it's not present in FF, are we still okay to add it?

Reply:
I did it anyway in https://github.com/gfx-rs/wgpu/pull/6814/commits/0bc74e2c736bf9e41e76b2fd831317f1262f67fb using [`VariantArray`](https://docs.rs/strum/latest/strum/derive.VariantArray.html). I will revert it if we do not want strum as dep.